rust_src/src/buffers.rs

```diff
@@ -360,4 +361,17 @@ fn set_buffer(buffer_or_name: LispObject) -> LispObject {
     buffer
 }
 
+/// No such buffer error.
+#[no_mangle]
+pub fn nsberror(spec: LispObject) {
+    if let Some(s) = spec.as_string() {
+        error!(
+            "No buffer named {}",
+            std::str::from_utf8(s.as_slice()).unwrap()
```

Comment:
Should that be made a function or method?

Reply:
I don't think this is a good fit for a method. We have our *_or_error methods and that would maybe confusing.

Reply:
This code says "s is a string, then slice it, turn it into utf8. Trust me" Which is a whole lot of 's is kinda but not really a string'.

Reply:
Especially since as an Emacs string it might *not* be utf8.

Reply:
So I use `from_utf8_lossy` ?

Reply:
That could work, yes. But really, in the end you'd want to insert the bytes literally, so that Emacs on the other end can display them as intended.

Reply:
It should be forbidden to use multibyte chars for buffer names ;)
I'll try to fix this.

Reply:
You mean multibyte chars in buffer names ?

Reply:
If birkenfeld is write that non UTF8 can slip through there should be a test for it.

Reply:
You're right, but wouldn't it be better to simply not allow creating those buffers ? And I already see the answer to my mail "The docstring does not say it's not allowed to use non UTF8 chars, so it was probably intended that way" ;)

Reply:
Maybe it would also be good to deny the creation of non UTF8 buffer names.

Reply:
That is probably impossible since buffer names may be allowed to be unibyte strings (haven't checked this).